### PR TITLE
Make FileRef u4

### PR DESCRIPTION
**Pull request by DarkDimius**
### Test plan

See included automated tests.

### CODE REVIEW

File: core/GlobalState.h [hunk 1]
```diff
@@ -102,8 +102,8 @@ class GlobalState final {
     NameRef enterNameUTF8(std::string_view nm);
     NameRef lookupNameUTF8(std::string_view nm) const;
 
-    NameRef lookupNameUnique(UniqueNameKind uniqueNameKind, NameRef original, u2 num) const;
-    NameRef freshNameUnique(UniqueNameKind uniqueNameKind, NameRef original, u2 num);
+    NameRef lookupNameUnique(UniqueNameKind uniqueNameKind, NameRef original, u4 num) const;
```
Comment (DarkDimius) [inline core/GlobalState.h]: This creates static inits per file

File: core/GlobalState.h [hunk 2]
```diff
@@ -69,7 +69,7 @@ class GlobalState final {
     SymbolRef enterTypeMember(Loc loc, SymbolRef owner, NameRef name, Variance variance);
     SymbolRef enterTypeArgument(Loc loc, SymbolRef owner, NameRef name, Variance variance);
     SymbolRef enterMethodSymbol(Loc loc, SymbolRef owner, NameRef name);
-    SymbolRef enterNewMethodOverload(Loc loc, SymbolRef original, core::NameRef originalName, u2 num,
+    SymbolRef enterNewMethodOverload(Loc loc, SymbolRef original, core::NameRef originalName, u4 num,
```
Comment (DarkDimius) [inline core/GlobalState.h]: given than unique names need to be u4 due to https://github.com/sorbet/sorbet/pull/2791/files#r395269511, made this also u4 for consistency

File: core/Loc.h [hunk 3]
```diff
@@ -29,7 +29,7 @@ CheckSize(LocOffsets, 6, 1);
 class Loc final {
     struct {
         LocOffsets offsets;
-        unsigned int fileRef : 16;
+        unsigned int fileRef : 24;
```
Comment (jez) [inline core/Loc.h]: This is not a u4. is that ok?

Comment (DarkDimius) [reply core/Loc.h]: It's enough for now, I'll followup with doing this properly soon

**Comment by DarkDimius** [comment]
sudo merge

**Pull request merged**